### PR TITLE
3460 - 0

### PR DESCRIPTION
**Pull request by AntonStashevski**
## Номер
3460

## Номер задания
0

## Ссылка на видео с демо
https://www.youtube.com/watch?v=84QQGqsZiGc

## Комментарии
Крч, я очень долго провозился с вашим заданием squash commits, но даже при малейшей перезаписи истории я не могу отправить запрос, так что вот, все остальное я сделал

### CODE REVIEW

**Comment by AntonStashevski** [comment]
Screenshot from 2019-07-08 22-03-43

File: 3460/0/triangle.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+# rubocop:disable CyclomaticComplexity
+# rubocop:disable PerceivedComplexity
+#
+def triangle(side_a, side_b, side_c)
+  raise TriangleError unless side_a.positive?
```
Comment (stahor) [inline 3460/0/triangle.rb]: is 10, -5, -20 valid triangle?

File: 3460/0/triangle.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  raise TriangleError unless side_a + side_b > side_c
```
Comment (stahor) [inline 3460/0/triangle.rb]: 5, 1, 4

File: 3460/0/triangle.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  return :equilateral if side_a == side_b && side_b == side_c
```
Comment (stahor) [inline 3460/0/triangle.rb]: 20, 10, 10

File: 3460/0/about_scoring_project.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+#
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+# rubocop:disable Lint/UselessAssignment
```
Comment (stahor) [inline 3460/0/about_scoring_project.rb]: not needed here

File: 3460/0/triangle.rb (hunk 1 above)
Comment (AntonStashevski) [reply 3460/0/triangle.rb]: fixed

File: 3460/0/triangle.rb (hunk 2 above)
Comment (AntonStashevski) [reply 3460/0/triangle.rb]: fixed

File: 3460/0/triangle.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+end
+
+def triangle_kind(side_a, side_b, side_c)
+  return :equilateral if (side_a == side_b) && (side_b == side_c)
+  return :isosceles if (side_a == side_b) || (side_b == side_c) || (side_c == side_a)
+
+  :scalene
+end
+
+def triangle(side_a, side_b, side_c)
```
[the remark this answers is not in the document]
Comment (AntonStashevski) [reply 3460/0/triangle.rb]: Screenshot from 2019-07-08 23-50-05


the dog begins to swear

File: 3460/0/about_scoring_project.rb (hunk 4 above)
Comment (AntonStashevski) [reply 3460/0/about_scoring_project.rb]: the dog begins to swear
Screenshot from 2019-07-08 23-50-05

File: 3460/0/triangle.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+# rubocop:disable CyclomaticComplexity
+# rubocop:disable PerceivedComplexity
+#
+
+def triangle(side_a, side_b, side_c)
+  variable_one = side_a + side_b
```
Comment (stahor) [inline 3460/0/triangle.rb]: N-naming
Hint: u can create additional methods

File: 3460/0/about_scoring_project.rb [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+def score(dice)
+  triple_scores = [1000, 200, 300, 400, 500, 600]
+  single_scores = [100, 0, 0, 0, 50, 0]
+  (1..6).inject(0) do |score, number|
+    count = dice.count(number)
+    score += triple_scores[number - 1] * (count / 3)
+    score += single_scores[number - 1] * (count % 3)
```
Comment (stahor) [inline 3460/0/about_scoring_project.rb]: even dog says what to do.
Read `inject` docs

Comment (AntonStashevski) [reply 3460/0/about_scoring_project.rb]: fixed

File: 3460/0/triangle.rb (hunk 6 above)
Comment (AntonStashevski) [reply 3460/0/triangle.rb]: if I understood you correctly, then I fixed

File: 3460/0/about_scoring_project.rb [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
+  triple_scores = [1000, 200, 300, 400, 500, 600]
+  single_scores = [100, 0, 0, 0, 50, 0]
+  (1..6).inject(0) do |score, number|
```
Comment (stahor) [inline 3460/0/about_scoring_project.rb]: read about `sum`

File: 3460/0/about_scoring_project.rb [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
+  triple_scores = [1000, 200, 300, 400, 500, 600]
```
Comment (stahor) [inline 3460/0/about_scoring_project.rb]: what about constants? In the current version you create these arrays every time when the method is called

File: 3460/0/about_scoring_project.rb [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  single_scores = [100, 0, 0, 0, 50, 0]
+  (1..6).inject(0) do |score, number|
+    score + triple_scores[number - 1] * (dice.count(number) / 3) + single_scores[number - 1] * (dice.count(number) % 3)
```
Comment (stahor) [inline 3460/0/about_scoring_project.rb]: well you can also move parts of the expression into separate methods

File: 3460/0/triangle.rb [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+def triangle_kind(side_a, side_b, side_c)
+  return :equilateral if (side_a == side_b) && (side_b == side_c)
+  return :isosceles if (side_a == side_b) || (side_b == side_c) || (side_c == side_a)
+
+  :scalene
+end
+
+def triangle(side_a, side_b, side_c)
+  if side_a + side_b <= side_c || side_b + side_c <= side_a || side_c + side_a <= side_b
```
Comment (stahor) [inline 3460/0/triangle.rb]: create `valid_triangle?` or `valid_triangle_sides?` for this check. And ensure that passed sides are positive numbers.

File: 3460/0/triangle.rb [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+
+def triangle_kind(side_a, side_b, side_c)
+  return :equilateral if (side_a == side_b) && (side_b == side_c)
+  return :isosceles if (side_a == side_b) || (side_b == side_c) || (side_c == side_a)
```
Comment (stahor) [inline 3460/0/triangle.rb]: LGTM.
You can also try to use `[side_a, side_b, side_c].uniq.size` as `case` argument. Or even replace case with hash.

File: 3460/0/triangle.rb (hunk 11 above)
Comment (AntonStashevski) [reply 3460/0/triangle.rb]: I added tests for everything

File: 3460/0/triangle.rb (hunk 12 above)
Comment (AntonStashevski) [reply 3460/0/triangle.rb]: I use .min .max

File: 3460/0/about_scoring_project.rb (hunk 10 above)
Comment (AntonStashevski) [reply 3460/0/about_scoring_project.rb]: I move parts into separate method

File: 3460/0/about_scoring_project.rb (hunk 9 above)
Comment (AntonStashevski) [reply 3460/0/about_scoring_project.rb]: I deleted the hash from my program

File: 3460/0/about_scoring_project.rb (hunk 8 above)
Comment (AntonStashevski) [reply 3460/0/about_scoring_project.rb]: add sum in my program

**Pull request closed without merging**